inst/private/cygpath.m

```diff
@@ -0,0 +1,40 @@
+%% Copyright (C) 2022 Alex Vong
+%%
+%% This file is part of OctSymPy.
+%%
+%% OctSymPy is free software; you can redistribute it and/or modify
+%% it under the terms of the GNU General Public License as published
+%% by the Free Software Foundation; either version 3 of the License,
+%% or (at your option) any later version.
+%%
+%% This software is distributed in the hope that it will be useful,
+%% but WITHOUT ANY WARRANTY; without even the implied warranty
+%% of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See
+%% the GNU General Public License for more details.
+%%
+%% You should have received a copy of the GNU General Public
+%% License along with this software; see the file COPYING.
+%% If not, see <https://www.gnu.org/licenses/>.
+
+%% -*- texinfo -*-
+%% @defun  cygpath ()
+%% Convert Windows native path to Cygwin POSIX-style path.
+%%
+%% @seealso{python_env_is_cygwin_like}
+%% @end defun
+
+function posix_path = cygpath (native_path)
+  %% FIXME: only allow safe characters inside "..."
+  if ~isempty (strfind (native_path, '"'))
+    error ('cygpath: native path %s must not contain "', native_path);
+  end
+
+  [status, out] = system (['cygpath -u "' native_path '"']);
+  if status ~= 0
+    error ('cygpath: cygpath exited with status %d', status);
+  end
+
+  %% validate output
+  assert (ischar (out) && logical (regexp (out, '^[^\r\n]+[\r]?[\n]$')));
```

Comment:
The second part of this condition might be too specific. I'm not sure if `system` should really add a new line to the end of the `out` string.
That might actually be a bug in Octave. (But I didn't come around to investigate.)

Maybe better check that the string after removal of the (potential) trailing newline is non-empty?

Reply:
> The second part of this condition might be too specific. I'm not sure if `system` should really add a new line to the end of the `out` string.
> That might actually be a bug in Octave. (But I didn't come around to investigate.)
>
I would expect a console program to append exactly 1 newline to the output path. Otherwise, it would be unreadable to human. (0 or 1 is plausible, > 1 would be a bug)

> Maybe better check that the string after removal of the (potential) trailing newline is non-empty?
>
Indeed, it must be chars according to documentation. We can validate the path at the end, which is actually safer.